contrib/mesos/pkg/scheduler/components/algorithm/algorithm.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package algorithm
```

Comment:
Also we will turn podscheduler into something more clever in not too far future. Think about the 50000 pod example in 2 minutes, in analogy to Verizon with Marathon. This will require a new algorithm architecture. I would rather not concentrate too much on the current podschedulers.